src/Logic/UsersHelper.php

```diff
@@ -67,36 +69,78 @@ public static function get_user( array $data ): WP_User|bool {
 	}
 
 	/**
-	 * Get a username that is not in use from a desired username.
+	 * Confirms that the given username (`wp_users`.`user_login`) is unused.
+	 *
+	 * @param string $username The username (`wp_users`.`user_login`) to check.
+	 * @param int    $exclude_user_id A user ID to exclude from the check.
+	 *
+	 * @return bool
+	 */
+	public static function is_username_unused( string $username, int $exclude_user_id = 0 ): bool {
+		// Check to see if raw $username is unused. Assumption is that all sanitation (if necessary under the context) has already been performed on $username.
+		// `get_user_by` is not used here because it does some sanitization (via `sanitize_user()`), as well as caching.
+		global $wpdb;
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users -- We need raw uncached query results.
+		$prepared_sql = $wpdb->prepare( "SELECT ID FROM $wpdb->users WHERE user_login = %s", $username );
+
+		if ( $exclude_user_id ) {
+			// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: $sql_prepared is a prepared statement.
+			$prepared_sql = $wpdb->prepare( "$prepared_sql AND ID <> %d", $exclude_user_id );
+		}
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.NotPrepared
+		$found_no_match = null === $wpdb->get_var( $prepared_sql );
+
+		if ( $found_no_match && has_filter( 'nmt_additional_unused_username_check' ) ) {
+			// We've confirmed that the $username is unused, but certain plugins might want to add their own logic to check for additional conditions.
+			$found_no_match = apply_filters( 'nmt_additional_unused_username_check', $username, $exclude_user_id );
+		}
+
+		return $found_no_match;
+	}
+
+	/**
+	 * Get a username (`wp_users`.`user_login`) that is not in use, starting with a desired username.
 	 *
 	 * If the desired username is in use, a counter will be appended to it until an unused username is found.
 	 *
-	 * @param string $desired_username Desired username.
+	 * @param string $desired_username Desired username (`wp_users`.`user_login`).
 	 *
 	 * @return string An unused username.
+	 *
+	 * @throws InvalidArgumentException If the desired username is empty.
 	 */
 	public static function get_unused_username( string $desired_username ): string {
-		$original_username = $desired_username;
-		$desired_username  = trim( $desired_username );
-		$max_length        = 60;
-		if ( strlen( $desired_username ) >= $max_length ) {
-			$desired_username = trim( mb_substr( $desired_username, 0, $max_length ) );
+		if ( empty( $desired_username ) ) {
+			throw new InvalidArgumentException( 'Desired username (`wp_users`.`user_login`) cannot be empty.' );
+		}
+
+		$original_user_login = $desired_username;
+
+		if ( is_email( $desired_username ) ) {
+			$desired_username = substr( $desired_username, 0, strpos( $desired_username, '@' ) );
```

Comment:
I think using `$desired_username = trim( mb_substr( $desired_username, 0, $max_length ) );` is better here. 

And maybe we should check if it's empty after the substr transformation?